File: app/src/main/java/com/xomena/cmpfutboltfe/FootballField.java
package com.xomena.cmpfutboltfe;

import android.os.Parcel;
import android.os.Parcelable;
import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;

/**
 * Created by andriy on 9/14/14.
 */
public class FootballField implements Parcelable {
    private String county;
    private String name;
    private String address;
    private String description;
    private String phone;
    private double lat;
    private double lng;
    private String type;
    private String placeId;
    private double accessLat;
    private double accessLng;

    /**
     * Constructor
     * @param js_val JSONArray from data source
     */
    public FootballField(JSONArray js_val){
      for(int i=0; i<js_val.length(); i++){
        if(!js_val.isNull(i)){
          try {
              switch (i) {
                  case 0:
                      this.county = js_val.getString(i);
                      break;
                  case 1:
                      this.name = js_val.getString(i);
                      break;
                  case 2:
                      this.address = js_val.getString(i);
                      break;
                  case 3:
                      this.description = js_val.getString(i);
                      break;
                  case 4:
                      this.phone = js_val.getString(i);
                      break;
                  case 5:
                      this.lat = js_val.getDouble(i);
                      break;
                  case 6:
                      this.lng = js_val.getDouble(i);
                      break;
                  case 7:
                      this.type = js_val.getString(i);
                      break;
                  case 8:
                      this.placeId = js_val.getString(i);
                      break;
                  case 9:
                      if (!js_val.isNull(i) && !"".equals(js_val.getString(i))){
                          this.accessLat = js_val.getDouble(i);
                      }
                      break;
                  case 10:
                      if (!js_val.isNull(i) && !"".equals(js_val.getString(i))){
                          this.accessLng = js_val.getDouble(i);
                      }
                      break;
                  default:
                      break;
              }
          } catch (JSONException ex){
              Log.e("[JSON Array]", "Football field constructor", ex);
          }
        }
      }
    }

    public FootballField(Parcel source) {
        this.county = source.readString();
        this.name = source.readString();
        this.address = source.readString();
        this.description = source.readString();
        this.phone = source.readString();
        this.lat = source.readDouble();
        this.lng = source.readDouble();
        this.type = source.readString();
        this.placeId = source.readString();
        this.accessLat = source.readDouble();
        this.accessLng = source.readDouble();
    }

    /**
     * Getters
     */
    public String getCounty(){
        return this.county;
    }

    public String getName(){
        return this.name;
    }

    public String getAddress(){
        return this.address;
    }

    public String getDescription(){
        return this.description;
    }

    public String getPhone(){
        return this.phone;
    }

    public double getLat(){
        return this.lat;
    }

    public double getLng(){
        return this.lng;
    }

    public String getType(){
        return this.type;
    }

    public String getPlaceId() {
        return this.placeId;
    }

    public double getAccessLat() {
        return this.accessLat;
    }

    public double getAccessLng() {
        return this.accessLng;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel p, int flags) {
        p.writeString(this.county);
        p.writeString(this.name);
        p.writeString(this.address);
        p.writeString(this.description);
        p.writeString(this.phone);
        p.writeDouble(this.lat);
        p.writeDouble(this.lng);
        p.writeString(this.type);
        p.writeString(this.placeId);
        p.writeDouble(this.accessLat);
        p.writeDouble(this.accessLng);
    }

    public static final Parcelable.Creator<FootballField> CREATOR = new Parcelable.Creator<FootballField>() {

        @Override
        public FootballField createFromParcel(Parcel source) {
            return new FootballField(source);
        }

        @Override
        public FootballField[] newArray(int size) {
            return new FootballField[size];
        }

    };

}

File: app/src/main/java/com/xomena/cmpfutboltfe/model/FootballField.kt
package com.xomena.cmpfutboltfe.model

import android.os.Parcelable
import android.util.Log
import kotlinx.parcelize.Parcelize
import org.json.JSONArray
import org.json.JSONException

@Parcelize
data class FootballField(
    val county: String = "",
    val name: String = "",
    val address: String = "",
    val description: String = "",
    val phone: String = "",
    val lat: Double = 0.0,
    val lng: Double = 0.0,
    val type: String = "",
    val placeId: String = "",
    val accessLat: Double = 0.0,
    val accessLng: Double = 0.0
) : Parcelable {

    companion object {
        private const val LOG_TAG = "[JSON Array]"

        @JvmStatic
        fun fromJSONArray(jsVal: JSONArray): FootballField {
            var county = ""
            var name = ""
            var address = ""
            var description = ""
            var phone = ""
            var lat = 0.0
            var lng = 0.0
            var type = ""
            var placeId = ""
            var accessLat = 0.0
            var accessLng = 0.0

            for (i in 0 until jsVal.length()) {
                if (!jsVal.isNull(i)) {
                    try {
                        when (i) {
                            0 -> county = jsVal.getString(i)
                            1 -> name = jsVal.getString(i)
                            2 -> address = jsVal.getString(i)
                            3 -> description = jsVal.getString(i)
                            4 -> phone = jsVal.getString(i)
                            5 -> lat = jsVal.getDouble(i)
                            6 -> lng = jsVal.getDouble(i)
                            7 -> type = jsVal.getString(i)
                            8 -> placeId = jsVal.getString(i)
                            9 -> {
                                if (!jsVal.isNull(i) && jsVal.getString(i).isNotEmpty()) {
                                    accessLat = jsVal.getDouble(i)
                                }
                            }
                            10 -> {
                                if (!jsVal.isNull(i) && jsVal.getString(i).isNotEmpty()) {
                                    accessLng = jsVal.getDouble(i)
                                }
                            }
                        }
                    } catch (ex: JSONException) {
                        Log.e(LOG_TAG, "Football field constructor", ex)
                    }
                }
            }

            return FootballField(
                county, name, address, description, phone,
                lat, lng, type, placeId, accessLat, accessLng
            )
        }
    }
}
